src/app/core/services/auth/auth.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from "@angular/core";
+import { HttpClient } from "@angular/common/http";
+import { BehaviorSubject, Observable } from "rxjs";
+import { map } from "rxjs/operators";
+import { IUser } from "./../../../shared/interfaces/user";
+import { apiUrl } from "./../../../../environments/environment";
+
+@Injectable({
+  providedIn: "root"
+})
+export class AuthService {
+  private currentUserSubject: BehaviorSubject<IUser>;
+  public currentUser: Observable<IUser>;
+
+  constructor(private http: HttpClient) {
+    this.currentUserSubject = new BehaviorSubject<IUser>(
+      JSON.parse(localStorage.getItem("currentUser"))
+    );
+    this.currentUser = this.currentUserSubject.asObservable();
+  }
+
+  public get currentUserValue(): IUser {
+    return this.currentUserSubject.value;
+  }
+
+  register(user: IUser) {
+    return this.http.post<any>(`${apiUrl}/auth/register`, user)
+    .pipe(
+      map(user => {
+        return user;
+      })
+    );
+  }
+
+  login(email, password) {
+    return this.http
+      .post<any>(`${apiUrl}/auth/login`, { email, password })
+      .pipe(
+        map(user => {
+          localStorage.setItem("currentUser", JSON.stringify(user));
+          this.currentUserSubject.next(user);
+          return user;
+        })
+      );
+  }
+
+  logout() {
+    localStorage.removeItem("currentUser");
```

Comment:
If you want, you cache loged user and other data what do you need to save in separate service, like 'data-store'. App to you